src/n-acd.c

```diff
@@ -360,12 +360,12 @@ _c_public_ int n_acd_new(NAcd **acdp, NAcdConfig *config) {
         acd->max_bpf_map = 8;
 
         r = n_acd_bpf_map_create(&acd->fd_bpf_map, acd->max_bpf_map);
-        if (r)
-                return r;
 
-        r = n_acd_bpf_compile(&fd_bpf_prog, acd->fd_bpf_map, (struct ether_addr*) acd->mac);
-        if (r)
-                return r;
+        if (acd->fd_bpf_map >= 0) {
+                r = n_acd_bpf_compile(&fd_bpf_prog, acd->fd_bpf_map, (struct ether_addr*) acd->mac);
+                if (r)
+                        fd_bpf_prog = -1;
+        }
```

Comment:
Same as above, I think this should handle a possible `N_ACD_E_DENIED`. So something like this:

```c
fd_bpf_prog = -1;
r = n_acd_bpf_map_create(...);
if (!r)
    r = n_acd_bpf_compile(...);
if (r && r != N_ACD_E_DENIED)
    return r;
```